crates/domain/src/services/user_service.rs

```diff
@@ -0,0 +1,46 @@
+use std::sync::Arc;
+
+use axum::async_trait;
+
+use crate::errors::{ApiResult, ApiError};
+use crate::models::user::User;
+use crate::repositories::user_repository::DynUserRepository;
+
+pub type DynUserService = Arc<dyn UserService + Send + Sync>;
+use bcrypt::hash;
+
+#[async_trait]
+pub trait UserService {
+    async fn user_login(&self, id: &str, password: &str) -> ApiResult<String>;
+    async fn user_signup(&self, id: &str, password: &str) -> ApiResult<User>;
+}
+
+#[derive(Clone)]
+pub struct DomainUserService {
+    pub user_repository: DynUserRepository,
+}
+
+impl DomainUserService {
+    pub fn new(user_repository: DynUserRepository) -> Self {
+        Self {
+            user_repository,
+        }
+    }
+}
+
+#[async_trait]
+impl UserService for DomainUserService {
+    async fn user_login(&self, id: &str, password: &str) -> ApiResult<String> {
+        Ok(self.user_repository.user_login(id.to_string(), password).await?)
```

Comment:
Repository is a collection of the domain object, and it is supposed to have methods to store or load domain object to/from the repository. You'll want to `get_user_by_id`, and the `User` object may implements the `verify` method, something like this:

```
let user = self.user_repository.get_user_by_id(id.to_string()).await?;
if user.verify(password) {
    Ok(token)
} else {
    Err(...)
}
```